account_invoice_production_lot/models/account_invoice.py

```diff
@@ -36,7 +36,8 @@ def _compute_prod_lots(self):
             if not line.order_line_ids:
                 return
             line.prod_lot_ids = self.mapped(
-                'order_line_ids.procurement_ids.move_ids.lot_ids')
+                'order_line_ids.procurement_ids.move_ids').filtered(
+                lambda move: move.state == 'done').lot_ids
```

Comment:
Note that if there are multiple done moves, this gets an error: you should retrieve `lot_ids` using `mapped`.
Moreover, maybe it is more readable if you split all this in a few lines.